juju/juju
# [JUJU-1931] Remove the centos7 hack and model app base not series in state

Remove the "centos7" hack used to model centos series. To do this, we now model os name and channel instead of series.
In state, this means that the application's series attribute can be removed, and the charm origin platform stores os version (eg "20.04/stable") instead of series. And so we also correctly store centos info - "centos/7" instead of "centos/centos7". 

In core, we can now remove the helper methods and hacks needed to transform the centos series.

Charm origin and platform are mandatory now, so we rely on those being present and remove the various checks to handle them not being specified,

For operator k8s charm, we no longer store the series at "kubernetes". We use the underlying ubuntu version like we do for sidecar charms (we use focal). So now charm base is properly surfaced everywhere.

Also remove all the 2.9 upgrade steps - these are no longer needed since 3.0 can only upgrade from 2.9.36 and at the time of writing, this is the latest 2.9 version. The steps were removed here to avoid having to waste time updating them to deal with the series->channel change.

## Checklist

- [x ] Code style: imports ordered, good names, simple structure, etc

## QA steps

deploy single machine and k8s charms
deploy machine and k8s bundles
check juju status

deploy slurmd (a centos 7 charm)

check that model migration works
